### PR TITLE
Add rake task for image optimization

### PR DESCRIPTION
**Pull request by dshevtsov**
## Purpose of this pull request

This pull request add a `rake image_opt` task that runs the [image_opt](https://github.com/toy/image_optim) tool for a list of modified files, or for the particular path if provided with `path` argument.

## Testing

Update your gem bundle with `bundle install` to add the tool

1. Add one or more images to the project
2. Run `rake image_optim`

Expected result: the new image/images are processed and compressed.

Example:

```
~/Projects/repos/magento/devdocs (ds_image-compr-plugin *)$ rake image_optim
bin/image_optim is not an image or there is no optimizer for it
Gemfile is not an image or there is no optimizer for it
Gemfile.lock is not an image or there is no optimizer for it
Rakefile is not an image or there is no optimizer for it
optimizing: 100.0% (elapsed: 24s)
48.60% 148.5K  Screen-Shot-2019-05-15-at-14.41.09.png
------         common/images/b2b/payment-on-credit.png
------         common/images/cloud/cloud-fastly-customize-503-response.png
Total: 48.58% 148.5K
```

3. Run `rake image_optim path=mftf` to test image compression at a particular directory

### CODE REVIEW

**Comment by shrielenee** [comment]
I have some questions about how to use this, but I suppose more info is forthcoming as it gets approved by others.

**Comment by dshevtsov** [comment]
@shrielenee, each time you add images to the repository, before you commit them, you should run `rake image_optim` to optimize the images before they become an integral part of git history.
Try to test it, and let me know if you have questions.

File: Gemfile [hunk 1]
```diff
@@ -12,6 +12,11 @@ group :test do
   gem 'launchy'
 end
 
+group :optimiztion do
```
[the remark this answers is not in the document]
Comment (dshevtsov) [reply Gemfile]: fixed

**Comment by dshevtsov** [comment]
running tests

**Comment by shrielenee** [comment]
> image_optim

Ok. Is this documented somewhere? I often find it hard to remember exactly what to add to my steps for tasks like running `rake`. If there is a page I can refer to remind myself that would make it easier.

**Comment by dshevtsov** [comment]
Good point. We've just discussed this on DocTechTalk and came up to the idea that we need one task that will run all the checks  before we commit the changes. This will be implemented as `rake check` and will check only modified files.

**Pull request merged**